components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -80,62 +87,65 @@ export function WorkspaceEntry({desc, model}: {desc: WorkspaceInfo, model: Works
                 setModalVisible(true);
             }
         }
-    ];
+    );
     const project = getProject(ws);
-    const startWsOnClick = (event: MouseEvent) => {
-        window.location.href = startUrl.toString();
-    }
     const showChanges = (event: MouseEvent) => {
+        event.preventDefault();
         setChangesModalVisible(true);
     }
-    return <div className="whitespace-nowrap flex space-x-2 py-6 px-6 w-full justify-between hover:bg-gray-100 cursor-pointer rounded-xl">
-        <div className="pr-3 self-center" onClick={startWsOnClick}>
-            <div className={stateClassName}>
-                &nbsp;
+    return <div>
+        <a className="rounded-xl whitespace-nowrap flex space-x-2 py-6 px-6 w-full justify-between hover:bg-gray-100 cursor-pointer" href={startUrl.toString()}>
+            <div className="pr-3 self-center">
+                <div className={stateClassName}>
+                    &nbsp;
+            </div>
+            </div>
+            <div className="flex flex-col w-3/12">
+                <div className="font-medium text-gray-800 truncate hover:underline">{ws.id}</div>
+                <a href={project ? 'https://' + project : undefined}><div className="text-sm overflow-ellipsis truncate text-gray-400 truncate">{project || 'Unknown'}</div></a>
             </div>
-        </div>
-        <div className="flex flex-col w-3/12" onClick={startWsOnClick}>
-            <div className="font-medium text-gray-800 truncate hover:underline">{ws.id}</div>
-            <a href={project ? 'https://'+ project : undefined}><div className="text-sm overflow-ellipsis truncate text-gray-400 truncate">{project || 'Unknown'}</div></a>
-        </div>
-        <div className="flex w-4/12 truncate overflow-ellipsis" onClick={startWsOnClick}>
-            <div className="flex flex-col">
-                <div className="font-medium text-gray-500 truncate">{ws.description}</div>
-                <div className="text-sm text-gray-400 truncate">{ws.contextURL}</div>
+            <div className="flex w-4/12 truncate overflow-ellipsis">
+                <div className="flex flex-col">
+                    <div className="font-medium text-gray-500 truncate">{ws.description}</div>
+                    <div className="text-sm text-gray-400 truncate">{ws.contextURL}</div>
+                </div>
+            </div>
+            <div className="flex w-2/12" onClick={numberOfChanges > 0 ? showChanges : undefined}>
+                <div className="flex flex-col">
+                    <div className="font-medium text-gray-500 truncate">{currentBranch}</div>
+                    {
+                        numberOfChanges > 0 ?
+                            <div className={"text-sm text-red truncate cursor-pointer hover:underline"} onClick={showChanges}>{changesLabel}</div>
```

Comment:
**question**: Changes currently open a modal. What do you think of having a dropdown here as seen in the specs? If you think a modal fits better, let me know so that I can provide some design specs.